Cython/Compiler/Optimize.py

```diff
@@ -2312,6 +2312,38 @@ def _optimise_generic_builtin_method_call(self, node, attr_name, function, arg_l
         return ExprNodes.CachedBuiltinMethodCallNode(
             node, function.obj, attr_name, arg_list)
 
+    PyObject_String_func_type = PyrexTypes.CFuncType(
+        Builtin.str_type, [
```

Comment:
@scoder

Is the commented out bit for `str` still true?
https://github.com/cython/cython/blob/5ab92ce56e3826bb0743b32b2f8a2270bf199222/Cython/Compiler/Builtin.py#L349-L355

I'm struggling to understand why it should be true - at least in Python 2.7 `PyObject_Str` looks pretty diligent at ensuring that it returns a string?

It seems that either the assignment of `Builtin.str_type` in this PR might be dubious, or (better) that `str` can be added back to `types_that_return_their_instance` which would be nice because it'd allow other things to use the type earlier.

Reply:
In Py2, `str()` is allowed to return a unicode object. So, yes, it should say `object` here. Thanks!